pkg/uninstaller/wait.go

```diff
@@ -0,0 +1,156 @@
+		return false, err
+	}
+	options := metav1.ListOptions{
+		LabelSelector: labelString,
+	}
+
+	objectList, err = client.Resource(objectsToCheck.gvr).Namespace("").List(context.TODO(), options)
+
+	if apierrors.IsNotFound(err) {
+		return true, nil
+	}
+
+	if err != nil {
+		return false, err
+	}
+
+	if len(objectList.Items) == 0 {
+		klog.V(6).Infof("%s not found", objectsToCheck.gvr)
+		return true, nil
+	}
+
+	return false, nil
+
+}
+
+// WaitForFinalizedUnjoin waits until all the incoming or outgoing peerings are deleted, by polling to ForeignCluster resource
+func WaitForFinalizedUnjoin(client dynamic.Interface, wg *sync.WaitGroup, result chan *resultType) {
+	defer wg.Done()
+	ticker := time.NewTicker(TickerInterval)
+	timeout := time.NewTicker(TickerTimeout)
+
+	quit := make(chan struct{})
+	var res = &resultType{
+		Success: false,
+	}
+
+	for {
+		select {
+		case <-timeout.C:
+			close(quit)
+		case <-ticker.C:
+			foreignClusterList, err := getForeignList(client)
+			if err != nil {
+				close(quit)
+			}
+
+			flag := checkPeeringsStatus(foreignClusterList)
+			if flag {
+				klog.Infof("All incoming or outgoing peering are disabled")
+				res.Success = true
+				close(quit)
+			}
+		case <-quit:
+			ticker.Stop()
+			timeout.Stop()
+			result <- res
+			return
+		}
+
+	}
+}
+
+// WaitForResources waits until existing peerings are disabled and associated resources are removed.
+func WaitForResources(client dynamic.Interface) error {
+	klog.Info("Waiting for Liqo Resources to be correctly deleted")
+	var wg sync.WaitGroup
+	DeletionResult := make(chan *resultType, len(toCheck))
+	ConditionResult := make(chan *resultType, ConditionsToCheck)
+	wg.Add(len(toCheck) + ConditionsToCheck)
+	for _, resource := range toCheck {
+		go WaitForEffectiveDeletion(client, resource, DeletionResult, &wg)
+	}
+	go WaitForFinalizedUnjoin(client, &wg, ConditionResult)
+	wg.Wait()
+
+	close(DeletionResult)
+	close(ConditionResult)
+
+	for elem := range DeletionResult {
+		if !elem.Success {
+			klog.Errorf("Error while waiting for %s to be deleted", elem.Resource.gvr.GroupResource())
+			return nil
+		}
+		printLabels, _ := generateLabelString(elem.Resource.labelSelector)
+		klog.Infof("%s instances with \"%s\" labels correctly deleted", elem.Resource.gvr.GroupResource(), printLabels)
+	}
+	for elem := range ConditionResult {
+		if !elem.Success {
+			klog.Errorf("All peerings have been correctly deleted")
+			return nil
+		}
+
+	}
+
+	return nil
+}
+
+// WaitForEffectiveDeletion waits until toCheck resources are deleted.
+func WaitForEffectiveDeletion(client dynamic.Interface, toCheck toCheckDeleted, result chan *resultType, wg *sync.WaitGroup) {
+	defer wg.Done()
+	ticker := time.NewTicker(TickerInterval)
+	timeout := time.NewTicker(TickerTimeout)
+	quit := make(chan struct{})
+	var wError error
+	value := false
+	var res = &resultType{
+		Resource: toCheck,
+		Success:  false,
+	}
+	for {
+		select {
+		case <-timeout.C:
```

Comment:
Is it possible to factorize this behavior with the very similar one a few lines before?